flink-end-to-end-tests/flink-datastream-allround-test/src/main/java/org/apache/flink/streaming/tests/DataStreamAllroundTestProgram.java

```diff
@@ -0,0 +1,304 @@
+
+	private static final ConfigOption<Boolean> STATE_BACKEND_FILE_ASYNC = ConfigOptions
+		.key("state_backend.file.async")
+		.defaultValue(true)
+		.withDescription("Activate or deactivate asynchronous snapshots if FileStateBackend is selected.");
+
+	private static final ConfigOption<Integer> SEQUENCE_GENERATOR_SRC_KEYSPACE = ConfigOptions
+		.key("sequence_generator_source.keyspace")
+		.defaultValue(1000);
+
+	private static final ConfigOption<Integer> SEQUENCE_GENERATOR_SRC_PAYLOAD_SIZE = ConfigOptions
+		.key("sequence_generator_source.payload_size")
+		.defaultValue(20);
+
+	private static final ConfigOption<Long> SEQUENCE_GENERATOR_SRC_SLEEP_TIME = ConfigOptions
+		.key("sequence_generator_source.sleep_time")
+		.defaultValue(0L);
+
+	private static final ConfigOption<Long> SEQUENCE_GENERATOR_SRC_SLEEP_AFTER_ELEMENTS = ConfigOptions
+		.key("sequence_generator_source.sleep_after_elements")
+		.defaultValue(0L);
+
+	private static final ConfigOption<Long> SEQUENCE_GENERATOR_SRC_EVENT_TIME_MAX_OUT_OF_ORDERNESS = ConfigOptions
+		.key("sequence_generator_source.event_time.max_out_of_order")
+		.defaultValue(500L);
+
+	private static final ConfigOption<Long> SEQUENCE_GENERATOR_SRC_EVENT_TIME_CLOCK_PROGRESS = ConfigOptions
+		.key("sequence_generator_source.event_time.clock_progress")
+		.defaultValue(100L);
+
+	// -----------------------------------------------------------------------------------------------------------------
+
+	public static void main(String[] args) throws Exception {
+		final ParameterTool pt = ParameterTool.fromArgs(args);
+
+		final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+
+		setupEnvironment(env, pt);
+
+		env.addSource(createEventSource(pt))
+			.assignTimestampsAndWatermarks(createTimestampExtractor(pt))
+			.keyBy(Event::getKey)
+			.map(createArtificialKeyedStateMapper(
+				// map function simply forwards the inputs
+				(MapFunction<Event, Event>) in -> in,
+				// state is updated per event as a wrapped ComplexPayload state object
+				(Event first, ComplexPayload second) -> new ComplexPayload(first), //
+				Arrays.asList(
+					new KryoSerializer<>(ComplexPayload.class, env.getConfig()))
+				)
+			)
+			.name("ArtificalKeyedStateMapper")
+			.returns(Event.class)
+			.keyBy(Event::getKey)
+			.flatMap(createSemanticsCheckMapper(pt))
+			.name("SemanticsCheckMapper")
+			.addSink(new PrintSinkFunction<>());
+
+		env.execute("General purpose test job");
+	}
+
+	public static void setupEnvironment(StreamExecutionEnvironment env, ParameterTool pt) throws Exception {
+
+		// set checkpointing semantics
+		String semantics = pt.get(TEST_SEMANTICS.key(), TEST_SEMANTICS.defaultValue());
+		long checkpointInterval = pt.getLong(ENVIRONMENT_CHECKPOINT_INTERVAL.key(), ENVIRONMENT_CHECKPOINT_INTERVAL.defaultValue());
+		CheckpointingMode checkpointingMode = semantics.equalsIgnoreCase("exactly-once")
```

Comment:
Maybe we could already catch any random string that does not represent a checkpointing semantic and throw an `IllegalArgumentException`?